mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,49 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+
+      //grab first active index and value by default
+      var maxIdx = indices(0)
+      var maxValue = values(0)
+
+      foreachActive { (i, v) =>
+        if (v > maxValue) {
+          maxIdx = i
+          maxValue = v
+        }
+      }
+
+      // look for inactive values incase all active node values are negative
+      if(size != values.size && maxValue <= 0){
```

Comment:
Sorry to be that guy, but it looks like this would also fail our current convention that the first idx should be returned,

if maxValues is zero and if the activeIndex that has a value zero is lesser than all inactive indices, something like.

    val a = SparseVector(3, Array(0), Array(0))

It seems that argmax would return 1 in this case. (correct me is I'm wrong)